hedera-node/src/main/java/com/hedera/services/state/virtual/schedule/ScheduleSecondVirtualValue.java

```diff
@@ -0,0 +1,249 @@
+
+	static final int CURRENT_VERSION = 1;
+
+	static final long RUNTIME_CONSTRUCTABLE_ID = 0x1d2377926e3a85fcL;
+
+	private final NavigableMap<RichInstant, ImmutableLongList> ids;
+
+	private boolean immutable;
+
+
+	public ScheduleSecondVirtualValue() {
+		this(TreeMap::new);
+	}
+
+	public ScheduleSecondVirtualValue(Map<RichInstant, ? extends LongList> ids) {
+		this();
+		ids.forEach((k, v) -> this.ids.put(k, v.toImmutable()));
+	}
+
+	private ScheduleSecondVirtualValue(Supplier<NavigableMap<RichInstant, ImmutableLongList>> ids) {
+		this.ids = ids.get();
+	}
+
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) {
+			return true;
+		}
+		if (o == null || ScheduleSecondVirtualValue.class != o.getClass()) {
+			return false;
+		}
+
+		var that = (ScheduleSecondVirtualValue) o;
+		return Objects.equals(this.ids, that.ids);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hashCode(ids);
+	}
+
+	@Override
+	public String toString() {
+		var helper = MoreObjects.toStringHelper(ScheduleSecondVirtualValue.class)
+				.add("ids", ids);
+		return helper.toString();
+	}
+
+	@Override
+	public void deserialize(SerializableDataInputStream in, int version) throws IOException {
+		int s = in.readInt();
+		ids.clear();
+		for (int x = 0; x <	s; ++x) {
+			int n = in.readInt();
+			LongArrayList l = new LongArrayList(n);
+			for (int y = 0; y <	n; ++y) {
+				l.add(in.readLong());
+			}
+			if (l.size() > 0) {
+				ids.put(RichInstant.from(in), l.toImmutable());
+			}
+		}
+	}
+
+	@Override
+	public void deserialize(ByteBuffer in, int version) throws IOException {
+		int s = in.getInt();
+		ids.clear();
+		for (int x = 0; x <	s; ++x) {
+			int n = in.getInt();
+			LongArrayList l = new LongArrayList(n);
+			for (int y = 0; y <	n; ++y) {
+				l.add(in.getLong());
+			}
+			if (l.size() > 0) {
+				ids.put(new RichInstant(in.getLong(), in.getInt()), l.toImmutable());
+			}
+		}
+	}
+
+	@Override
+	public void serialize(SerializableDataOutputStream out) throws IOException {
+		out.writeInt(ids.size());
+		for (var e : ids.entrySet()) {
+			out.writeInt(e.getValue().size());
+			for (int x = 0; x < e.getValue().size(); ++x) {
+				out.writeLong(e.getValue().get(x));
+			}
+			e.getKey().serialize(out);
+		}
+	}
+
+	@Override
+	public void serialize(ByteBuffer out) throws IOException {
+		out.putInt(ids.size());
+		for (var e : ids.entrySet()) {
+			out.putInt(e.getValue().size());
+			for (int x = 0; x < e.getValue().size(); ++x) {
+				out.putLong(e.getValue().get(x));
+			}
+			out.putLong(e.getKey().getSeconds());
+			out.putInt(e.getKey().getNanos());
+		}
+	}
+
+	@Override
+	public long getClassId() {
+		return RUNTIME_CONSTRUCTABLE_ID;
+	}
+
+	@Override
+	public int getVersion() {
+		return CURRENT_VERSION;
+	}
+
+	@Override
+	public ScheduleSecondVirtualValue copy() {
+		var fc = new ScheduleSecondVirtualValue(ids);
+
+		this.setImmutable(true);
+
+		return fc;
+	}
+
+	public void add(RichInstant instant, LongList idList) {
```

Comment:
The second argument is a `LongList` instead of `long` just to future-proof for a transaction type that schedules multiple transactions at once?

A small javadoc that clarifies why this is a `LongList` might be nice.

Reply:
Yeah, it is possible to schedule more than one thing for the same second.